gazebo_ros/test/disable_perf_metrics.test.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import launch
+import launch.actions
+
+import launch_testing
+import launch_testing.asserts
+import pytest
+import os
+
+import rclpy
+from gazebo_msgs.msg import PerformanceMetrics
+from rclpy.node import Node
+from rcl_interfaces.msg import Parameter, ParameterValue
+from ros2param.api import call_set_parameters
+import time
+from threading import Thread
+
+
+@pytest.mark.launch_test
+def generate_test_description():
+    os.chdir(os.path.dirname(__file__))
+    return launch.LaunchDescription([
+        launch.actions.ExecuteProcess(
+            cmd=['gzserver', '-s', 'libgazebo_ros_init.so'],
+            output='screen',
+            sigterm_timeout='20',
+            sigkill_timeout='20'
+        ),
+        launch_testing.actions.ReadyToTest()
+    ])
+
+
+class MyTestingClass(unittest.TestCase):
+    def test_routine(self):
+        rclpy.init()
+        node = MakeTestNode()
+        node_found = node.wait_for_gazebo_node()
+
+        if node_found:
+            response = node.set_parameter(False)
+            if response.successful:
+                print("enable_performance_metrics successfully set to false !")
+                node.start_subscriber()
+                time.sleep(2.5)
+
+                if node.msg_count > 0:
+                    print("Received {} messages, test failed".format(node.msg_count))
+                    rclpy.shutdown()
+                    assert False
+                else:
+                    print("No messages were published")
+                    rclpy.shutdown()
+                    assert True
+            else:
+                print("Parameter could not be set, exiting")
+                rclpy.shutdown()
+                assert False
+        else:
+            print("Could not find Gazebo node, exiting")
+            rclpy.shutdown()
+            assert False
+
+
+class MakeTestNode(Node):
+    def __init__(self):
+        super().__init__('testing_node')
+        self.flag_gazebo_node = False
+        self.msg_count = 0
+
+    def wait_for_gazebo_node(self):
```

Comment:
It might be nice to pass the timeout as a parameter, e.g.
```suggestion
    def wait_for_gazebo_node(self, timeout_sec=5.0):
```